packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -0,0 +1,93 @@
+import { Aliases, AuthInfo } from '@salesforce/core';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { ForceConfigGet } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+
+export async function setupWorkspaceOrgType(isActivation?: boolean) {
+  const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  let isScratchOrg = false;
+  if (defaultUsernameIsSet) {
+    const username = await getUsername(defaultUsernameOrAlias!);
+    try {
+      isScratchOrg = await isAScratchOrg(username);
+    } catch (e) {
+      if (e.name === 'NamedOrgNotFound') {
+        // TODO: what should we do in this case? Expose all source commands by default?
```

Comment:
What should we do if a default workspace username is set, but no AuthInfo file is found for the username in the file system? For now, I am exposing all of the source commands in this case.

Reply:
Yeah, this is the only reasonable path since we have no other info to decide otherwise.
The other extreme alternative is to not expose anything, which is not acceptable.